vxquery-rest/src/test/java/org/apache/vxquery/rest/AbstractRestServerTest.java

```diff
@@ -0,0 +1,139 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.vxquery.rest;
+
+import org.apache.htrace.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.HttpEntity;
+import org.apache.hyracks.control.cc.ClusterControllerService;
+import org.apache.hyracks.control.common.controllers.CCConfig;
+import org.apache.hyracks.control.common.controllers.NCConfig;
+import org.apache.hyracks.control.nc.NodeControllerService;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.StringReader;
+import java.net.InetAddress;
+import java.nio.file.Files;
+
+import static org.apache.vxquery.rest.Constants.HttpHeaderValues.CONTENT_TYPE_JSON;
+import static org.apache.vxquery.rest.Constants.HttpHeaderValues.CONTENT_TYPE_XML;
+
+public class AbstractRestServerTest {
+
+    private static ClusterControllerService clusterControllerService;
+    private static NodeControllerService nodeControllerService;
+
+    protected static VXQueryApplication application;
+    protected static RestServer restServer;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        System.setProperty(Constants.Properties.VXQUERY_PROPERTIES_FILE, "src/test/resources/vxquery.properties");
+
+        startLocalHyracks();
+        CCConfig ccConfig = clusterControllerService.getCCConfig();
+        System.setProperty(Constants.Properties.HYRACKS_CLIENT_IP, ccConfig.clientNetIpAddress);
+        System.setProperty(Constants.Properties.HYRACKS_CLIENT_PORT, String.valueOf(ccConfig.clientNetPort));
+
+        application = new VXQueryApplication();
+        application.start();
+        restServer = application.getRestServer();
+    }
+
+    /**
+     * Start local virtual cluster with cluster controller node and node controller nodes. IP address provided for node
+     * controller is localhost. Unassigned ports 39000 and 39001 are used for client and cluster port respectively.
+     * Creates a new Hyracks connection with the IP address and client ports.
+     *
+     * @throws Exception
+     */
+    private static void startLocalHyracks() throws Exception {
+        String localAddress = InetAddress.getLocalHost().getHostAddress();
+        CCConfig ccConfig = new CCConfig();
+        ccConfig.clientNetIpAddress = localAddress;
+        ccConfig.clientNetPort = 39000;
+        ccConfig.clusterNetIpAddress = localAddress;
+        ccConfig.clusterNetPort = 39001;
+        ccConfig.httpPort = 39002;
+        ccConfig.profileDumpPeriod = 10000;
+        clusterControllerService = new ClusterControllerService(ccConfig);
```

Comment:
I'm not sure what version of Hyracks you're on but you should be able to add the VXQueryApplication here either as a 2nd parameter to ClusterControllerService or as part of the CCConfig and then it will be started when you call clusterControllerService.start()